### PR TITLE
RPF-97: support order fill side

### PR DESCRIPTION
**Pull request by ex1st0r**

### CODE REVIEW

File: packages/sdk/src/order/encode-data.ts [hunk 1]
```diff
@@ -4,6 +4,17 @@ import type { Ethereum } from "@rarible/ethereum-provider"
 //todo wrongEncode когда применять?
 export function encodeData(ethereum: Ethereum, data: OrderData, wrongEncode: Boolean = false): [string, string] {
 	switch (data.dataType) {
+		case "RARIBLE_V2_DATA_V2": {
+			const encoded = ethereum.encodeParameter(DATA_V2_TYPE, {
+				payouts: data.payouts,
+				originFees: data.originFees,
+				isMakeFill: data.isMakeFill,
+			})
+			if (wrongEncode) {
```
Comment (evgenynacu) [inline packages/sdk/src/order/encode-data.ts:13]: Here this logic is not needed

File: packages/sdk/src/order/fill-order/rarible-v2.ts [hunk 2]
```diff
@@ -25,10 +25,25 @@ export class RaribleV2OrderHandler implements OrderHandler<RaribleV2OrderFillReq
 
 	invert(request: RaribleV2OrderFillRequest, maker: Address): SimpleRaribleV2Order {
 		const inverted = invertOrder(request.order, request.amount, maker)
-		inverted.data = {
-			dataType: "RARIBLE_V2_DATA_V1",
-			originFees: request.originFees || [],
-			payouts: request.payouts || [],
+		switch (request.order.data.dataType) {
+			case "RARIBLE_V2_DATA_V1": {
```
Comment (evgenynacu) [inline packages/sdk/src/order/fill-order/rarible-v2.ts:29]: We can create v2 every time
Can this simplify things a bit? We should calculate isMakeFill correctly

File: packages/sdk/src/order/encode-data.ts (hunk 1 above)
Comment (ex1st0r) [reply packages/sdk/src/order/encode-data.ts]: done

File: packages/sdk/src/order/fill-order/rarible-v2.ts (hunk 2 above)
Comment (ex1st0r) [reply packages/sdk/src/order/fill-order/rarible-v2.ts:29]: RARIBLE_V2_DATA_V1 processing is required to fill in orders that have already been created. New orders are created with RARIBLE_V2_DATA_V2

**Pull request merged**